File: Lua/src/apuntes/strings.lua
-- Ejemplos de manejo de Strings

-- Primero crearemos una String:
local nombre = "candela"


-- Convierte una String a mayúsculas
print(string.upper(nombre))


-- Convierte una String a minúsculas
print(string.lower(nombre))


-- Reemplaza una porción de la String por un valor dado.
print(string.gsub(nombre, "la", "", 1))


-- Invierte el orden de los caracteres de una String
print(string.reverse(nombre))


-- Repite una String una cantidad determinada de veces
print(string.rep(nombre, 2))


-- Cuenta la cantidad de caracteres de una String
print(string.len(nombre))
-- O también
print(#nombre)


-- Devuelve el caracter al cual le pertenece el código ASCII dado
print(string.char(65))


-- Devuelve el código ASCII que le pertenece al caracter dado
print(string.byte("A"))


-- EXTRA
-- Convierte la primera letra de una String en mayúscula
print(string.gsub(string.gsub(nombre, "%a", string.lower), "^%l", string.upper))

File: Python/src/apuntes/strings.py
# Ejemplos de manejo de Strings

# Primero crearemos una String:
nombre = "candela"


# Convierte la primera letra de una String en mayúscula
print(nombre.capitalize())


# Convierte una String a mayúsculas
print(nombre.upper())


# Convierte una String a minúsculas
print(nombre.lower())


# Convierte la primera letra de cada palabra en mayúscula
print(nombre.title())


# Devuelve el número de veces que una determinada letra aparece en una String
print(nombre.count("a"))


# Devuelve TRUE si la String termina en un valor específico
print(nombre.endswith("a"))


# Busca una String dada dentro de otra. Si la encuentra, devuelve la posición de la misma, si no la encuentra, devuelve -1
print(nombre.find("de"))


# Busca un caracter (letra) dada dentro de una String. Si la encuentra, devuelve la posición de la misma, si no la encuentra, devuelve un error.
print(nombre.index("e"))


# Devuelve una String con el valor dado reemplazado por un valor específico (una letra o una palabra)
print(nombre.replace("candela", "luz"))